x-pack/plugins/security/server/prompt_page.tsx

```diff
@@ -54,6 +55,14 @@ export function PromptPage({
     `${basePath.serverBasePath}/ui/legacy_light_theme.css`,
   ];
 
+  // The only concern for style tag order is that EUI global styles precede compiled Sass.
+  // The following will have Emotion styles prepend the <head> and load first.
+  const emotionCache = createCache({
```

Comment:
__issue:__ it looks this change doesn't play well with this page, but I'm not sure why exactly, do you have an idea (see screenshot below, to test just create a user without any roles and try to log in to Kibana)? Also, just for my understanding, why don't we need both `global` and `default` caches here like in other root level pages (and `meta` placeholders in HTML as well)?

![image](https://user-images.githubusercontent.com/1713708/175503919-7bff61bf-cf6c-4730-b349-7e805cab51b1.png)

Reply:
Doh! @thompsongl any thoughts here?

Reply:
> Also, just for my understanding, why don't we need both global and default caches here like in other root level pages (and meta placeholders in HTML as well)?

To try and answer this (Greg, correct me if I'm wrong) - we didn't bother adding them in here because these are super basic pages that aren't using utility classes (which is the main thing that in this PR that we're adding out of order/after SCSS). Without the need for a cache that comes after our Sass/CSS, all EUI Emotion styles/CSS can come before the Sass imports (which is what the `prepend` configuration should be doing).

I'm testing this locally now to try and figure out why this is breaking!

Reply:
So, I have a mostly-working fix for this in https://github.com/elastic/kibana/pull/134919/commits/36700d174cfca7557817b2fe7140598bfcdfc75a. it was the `.compat = true` line causing issues as I suspected it might on SSR.

Styling should at least be working now, but @thompsongl I might need to chat/pair with you more, because per [Emotion's SSR documentation](https://emotion.sh/docs/ssr) SSR styles get input into the `<body>`, _not_ the `<head>` tag:

<img width="800" alt="" src="https://user-images.githubusercontent.com/549407/175617749-eccd614c-4781-4bf4-bba7-49f2bc276f38.png">

This means that EUI's global CSS will override any Sass styles being imported. This honestly might be fine until we're on 100% Emotion and 0% Sass, _if_ the SSR page(s) in question aren't doing any super custom styling.

If not, then we need to follow Emotion's SSR documentation and import `@emotion/server` in order to correctly generate styles in the `<head>`: https://github.com/constancecchen/kibana/compare/eui-60.0.0...eui-emotion-ssr

Also another caveat is that unfortunately Emotion emits a lot of annoying console logs about `:first-child`/`:nth-child` selectors on SSR :|

<img width="600" alt="" src="https://user-images.githubusercontent.com/549407/175618206-37c25c7a-e932-406f-a8b0-54c9400e1d6f.png">

Reply:
> To try and answer this (Greg, correct me if I'm wrong) - we didn't bother adding them in here because these are super static/basic pages that aren't using utility classes (which is the main thing in other places that we're adding out of order/after SCSS). Without the need for a cache that comes after our Sass/CSS, all EUI Emotion styles/CSS can come before the Sass imports (which is what the prepend configuration should be doing).

Got it, thank you for clarifying!

> if the SSR page(s) in question aren't doing any super custom styling.

Yeah, these pages are supposed to be pretty simple and rely on standard components/styling.

Reply:
> follow Emotion's SSR documentation and import `@emotion/server` in order to correctly generate styles in the `<head>`

This is probably the right answer.

> Also another caveat is that unfortunately Emotion emits a lot of annoying console logs about :first-child/:nth-child selectors on SSR

And unfortunately we have no way to stop/hide these

Reply:
After some discussion offline with Greg, we'll move forward with EUI Emotion styles in the `<body>` for now (since it's not affecting rendering of the server pages that we can see) and look into adding the `@emotion/server` dependency in a separate PR, in order to reduce thrash.

@azasypkin do you mind re-checking the security server-side pages again and confirming that they look good to you, and approving if so?

Reply:
Looks good to me, thanks for making these changes!

Reply:
Wahoo! Thanks so much for catching the issue @azasypkin!